connection.go

```diff
@@ -1163,3 +1240,107 @@ func (conn *Connection) NewStream() (*Stream, error) {
 		Conn: conn,
 	}, nil
 }
+
+// checkProtocolVersion checks that expected protocol version is supported.
+func checkProtocolVersion(expected ProtocolVersion, actual ProtocolVersion) error {
+	if expected > actual {
+		return fmt.Errorf("protocol version %d is not supported", expected)
+	}
+
+	return nil
+}
+
+// checkProtocolFeatures checks that expected protocol features are supported.
+func checkProtocolFeatures(expectedList []ProtocolFeature, actualList []ProtocolFeature) error {
+	var found bool
+	var missingList []ProtocolFeature
+	// It seems that iterating over a small list is way faster
+	// than building a map: https://stackoverflow.com/a/52710077/11646599
+	for _, expected := range expectedList {
+		found = false
+		for _, actual := range actualList {
+			if expected == actual {
+				found = true
+			}
+		}
+		if !found {
+			missingList = append(missingList, expected)
+		}
+	}
+
+	if len(missingList) == 1 {
+		return fmt.Errorf("protocol feature %s is not supported", missingList[0])
+	} else if len(missingList) > 1 {
+		var sarr []string
+		for _, missing := range missingList {
+			sarr = append(sarr, missing.String())
+		}
+		return fmt.Errorf("protocol features %s are not supported", strings.Join(sarr, ", "))
+	}
+
+	return nil
+}
+
+// identify sends info about client protocol, receives info
+// about server protocol in response and store it in the connection.
+func (conn *Connection) identify(w *bufio.Writer, r *bufio.Reader) error {
+	var ok bool
+
+	werr := conn.writeIdRequest(w, conn.ClientProtocolVersion(), conn.ClientProtocolFeatures())
+	if werr != nil {
+		return werr
+	}
+
+	resp, rerr := conn.readIdResponse(r)
+	if rerr != nil {
+		if resp.Code == ErrUnknownRequestType {
+			// IPROTO_ID requests are not supported by server.
+			return nil
+		}
+
+		return rerr
+	}
+
+	if len(resp.Data) == 0 {
+		return fmt.Errorf("identify: unexpected response: no data")
+	}
+
+	conn.serverProtocolInfo, ok = resp.Data[0].(ProtocolInfo)
+	if !ok {
+		return fmt.Errorf("identify: unexpected response: wrong data")
+	}
+
+	return nil
+}
+
+// ServerProtocolVersion returns protocol version supported by
+// connected Tarantool server. Beware that values might be outdated
+// if connection is in a disconnected state.
+// Since 1.10.0
+func (conn *Connection) ServerProtocolVersion() ProtocolVersion {
+	return conn.serverProtocolInfo.Version
+}
```

Comment:
It may be appropriate to merge ServerProtocolVersion() + ServerProtocolFeatures() = ServerProtocolInfo() and ClientProtocolVersion() + ClientProtocolFeatures() = ClientProtocolInfo()

Reply:
Fixed